### PR TITLE
Get vendor

### PR DESCRIPTION
**Pull request by azharmalik3**
@mosic

### CODE REVIEW

**Comment by jgarciasacristan** [comment]
Looks ok to me. But just to say something, show method in VendorController can be slightly refactored:


```
def show(conn, %{"id" => id}) do
    case Vendor.by_exid(id) do 
      nil ->
       conn
         |> put_status(404)
         |> render(ErrorView, "error.json", %{message: "Vendor not found."})
      vendor ->
        conn
        |> render(VendorView, "show.json", %{vendor: vendor})
    end
end
```

File: web/controllers/vendor_controller.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      nil ->
+        conn
+        |> put_status(404)
+        |> render(ErrorView, "error.json", %{message: "Vendor not found."})
+      vendor ->
+        conn
+        |> render(VendorView, "show.json", %{vendor: vendor})
+    end
+  end
+end
```
Comment (mosic) [inline web/controllers/vendor_controller.ex]: Add a newline character.

File: web/views/vendor_view.ex [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  defp vendor_attributes(vendor) do
+    %{
+      id: vendor.exid,
+      name: vendor.name,
+      known_macs: vendor.known_macs,
+      total_models: Vendor.get_models_count(vendor),
+      logo: "https://evercam-public-assets.s3.amazonaws.com/#{vendor.exid}/logo.jpg",
+    }
+  end
+end
```
Comment (mosic) [inline web/views/vendor_view.ex]: Add a newline character.

File: web/views/vendor_view.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  def render("show.json", %{vendor: vendor}) do
+    %{vendors: render_many([vendor], __MODULE__, "vendor.json")}
+  end
+
+  def render("vendor.json", %{vendor: vendor}) do
+    vendor_attributes(vendor)  	
+  end
+
+  defp vendor_attributes(vendor) do
```
Comment (mosic) [inline web/views/vendor_view.ex]: There is no need for this function, put the struct in `render("vendor.json", ...)`.

File: web/controllers/vendor_controller.ex [hunk 4]
```diff
@@ -0,0 +1,17 @@
+defmodule EvercamMedia.VendorController do
+  use EvercamMedia.Web, :controller
+  alias EvercamMedia.VendorView
+  alias EvercamMedia.ErrorView
+
+  def show(conn, %{"id" => id}) do
```
Comment (mosic) [inline web/controllers/vendor_controller.ex]: The argument should be `exid`, not `id`.

File: web/router.ex [hunk 5]
```diff
@@ -41,6 +41,8 @@ defmodule EvercamMedia.Router do
     get "/cameras/:id/touch", CameraController, :update
     get "/cameras/:id/thumbnail/:timestamp", CameraController, :thumbnail
 
+    get "vendors/:id", VendorController, :show
```
Comment (mosic) [inline web/router.ex]: Add a slash before "vendors", so it looks like all other routes.

File: web/views/vendor_view.ex [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    %{
+      id: vendor.exid,
+      name: vendor.name,
+      known_macs: vendor.known_macs,
```
Comment (mosic) [inline web/views/vendor_view.ex]: This returns `known_macs: [""]` when there are no known macs, it should be just `known_macs: []`.

File: web/controllers/vendor_controller.ex (hunk 1 above)
Comment (azharmalik3) [inline web/controllers/vendor_controller.ex]: Newline added.

File: web/views/vendor_view.ex (hunk 2 above)
Comment (azharmalik3) [inline web/views/vendor_view.ex]: Newline added.

**Comment by azharmalik3** [comment]
@mosic Completed all your commented changes and ready to merge.

**Comment by mosic** [comment]
Nice work @azharmalik3, pulled into master.

I mentioned this before, when updating the pull request you should squash all changes to a single commit so unnecessary changes are left out. I did it just now for this pull request, here is the end of result: https://github.com/evercam/evercam-media/commit/33b1c10ae36cd83fb6b7b3f2e2684ba9dc41c45c.

**Pull request closed without merging**